contracts/src/protocol/PortalRegistry.sol

```diff
@@ -0,0 +1,100 @@
+// SPDX-License-Identifier: GPL-3.0-only
+pragma solidity =0.8.24;
+
+import { Ownable } from "@openzeppelin/contracts/access/Ownable.sol";
+import { XRegistry } from "./XRegistry.sol";
+import { Predeploys } from "../libraries/Predeploys.sol";
+import { XRegistryNames } from "../libraries/XRegistryNames.sol";
+
+/**
+ * @title PortalRegistry
+ * @notice Registry for OmniPortal deployments. Predeployed on Omni's EVM.
+ * @dev Using Ownable, rather than OwnableUpgradeable, because predeploys are initialized at genesis.
+ */
+contract PortalRegistry is Ownable {
+    /**
+     * @notice Emitted when a new OmniPortal deployment is registered.
+     */
+    event PortalRegistered(uint64 indexed chainId, address indexed addr, uint64 deployHeight, string finalizationStrat);
+
+    /**
+     * @notice The XRegistry predeploy.
+     */
+    XRegistry public constant xregistry = XRegistry(Predeploys.XRegistry);
+
+    /**
+     * @notice Portal deployments by chain ID.
+     */
+    mapping(uint64 => Deployment) public deployments;
+
+    /**
+     * @notice Deployment information for an OmniPortal.
+     * @custom:field chainId            The chain ID of the deployment.
+     * @custom:field addr               The address of the deployment.
+     * @custom:field deployHeight       The height at which the deployment was deployed.
+     * @custom:field finalizationStrat  The finalization strategy of the chain (e.g. "finalized")
+     */
+    struct Deployment {
+        uint64 chainId;
+        address addr;
+        uint64 deployHeight;
+        string finalizationStrat;
+    }
+
+    /**
+     * @notice Get the OmniPortal deployment for a chain.
+     */
+    function get(uint64 chainId) external view returns (Deployment memory) {
```

Comment:
why can't we just use the mapping directly?

Reply:
the struct?

Reply:
We can - I just liked naming of `get` and `list`